### PR TITLE
polishing timers and log_dist

### PR DESCRIPTION
**Pull request by clumsy**
Here's a list of proposed changes:
- shortening timers' names, e.g. `forward` -> `fwd`, `backward` -> `bwd`, `alltoall` -> `a2a`
- removing redundant `rank=0` in `log_dist` section as it is already present (`[Rank 0]`)
- using lower case for all timers' names
- (re)using constant for all timers
- removing seemingly redundant `backward_tied_allreduce_microstep`
- removing repetitive `start_timers`, `stop_timers` and `log_timers`, switching to `timer.log`
- removing `timer_names` default `set()` since default values have global scope

E.g.:
```
[2023-07-19 14:09:21,612] [INFO] [logging.py:96:log_dist] [Rank 0] rank=0 time (ms) | forward_microstep: 2195.18 | backward_microstep: 443.16 | backward_inner_microstep: 206.54 | backward_allreduce_microstep: 236.57 | step_microstep: 0.12
```

becomes:
```
[2023-07-19 14:09:21,612] [INFO] [logging.py:96:log_dist] [Rank 0] time (ms) | fwd_microstep: 2195.18 | bwd_microstep: 443.16 | bwd_inner_microstep: 206.54 | bwd_allreduce_microstep: 236.57 | step_microstep: 0.12
```

Open questions:
1. Not sure if it's worth shortening `microstep` as `ustep`
1. Not sure if it's worth shortening `allreduce` as `ar`
1. Not sure if it's worth shortening `optimizer` as `opt`
1. Not sure if it's worth shortening `allgather` as `ag`

### CODE REVIEW

**Comment by tjruwase** [comment]
> For your consideration, @tjruwase

Thanks for the PR. I suspect this might break some log parsing scripts :).  But this is good. Will review asap.

**Comment by clumsy** [comment]
I see, one of the goals of this PR was to minimize the log line and fit more information (using the same abbreviations we use elsewhere, e.g. in FLOPS profiler). But I can change the timers' names back if this has become a contract for DeepSpeed and keep the rest of the changes :)

E.g. one particular thing I realized is that even though both FLOPS profiler and wall clock breakdown and not enabled the global timers are still on. That perhaps is unnecessary.

**Comment by tjruwase** [comment]
> I see, one of the goals of this PR was to minimize the log line and fit more information (using the same abbreviations we use elsewhere, e.g. in FLOPS profiler). But I can change the timers' names back if this has become a contract for DeepSpeed and keep the rest of the changes :)

No, this PR is fine; I was just kidding. I am not aware of any such contract. I think brevity is always good, such as fitting more stuff on my debug screen :).

File: deepspeed/runtime/engine.py [hunk 1]
```diff
@@ -1823,7 +1815,7 @@ def print_forward_breakdown(self, fwd_time):
 
         # if deepspeed.comm.get_rank() == 0:
         log_dist(
-            f"rank={dist.get_rank()} time (ms) | forward: {fwd_time:.2f} (forward_moe: {moe_time:.2f}, 1st alltoall: {falltoall:.2f}, 2nd alltoall: {salltoall:.2f}, top-k: {gate_time:.2f})",
+            f"time (ms) | fwd: {fwd_time:.2f} (fwd_moe: {moe_time:.2f}, 1st_a2a: {falltoall:.2f}, 2nd_a2a: {salltoall:.2f}, top_k: {gate_time:.2f})",
```
Comment (clumsy) [inline deepspeed/runtime/engine.py:1818]: Perhaps worth using timer constants here.

**Comment by samadejacobs** [comment]
@clumsy, thank you for this nice PR, we will kickstart review soon! As to your open questions, we can have `microstep` shortened to `ustep`, the rest are fine as they are.

**Comment by clumsy** [comment]
Are there any blocking comments? @samadejacobs, @tjruwase

**Review by samadejacobs — APPROVED** [review]
LGTM

**Pull request merged**